src/yunohost/certificate.py

```diff
@@ -639,6 +639,15 @@ def _prepare_certificate_signing_request(domain, key_file, output_folder):
     # Set the domain
     csr.get_subject().CN = domain
 
+    # Include xmpp-upload subdomain in subject alternate names
+    subdomain="xmpp-upload." + domain
+    try:
+        _check_domain_is_ready_for_ACME(subdomain)
```

Comment:
Here I think we just want to check the IP with something like 

```
if not _dns_ip_match_public_ip(get_public_ip(), subdomain):
    [...]
```

reason for this is that the other check that `_check_domain_is_ready_for_ACME` (`_domain_is_accessible_through_HTTP`) sometimes have false-negative in the context of hairpinning, which therefore would trigger an exception here. But anyway if we're in this function (`_prepare_certificate_signing_request`), we're already in a context where the domain is ready for ACME and we just want to check the DNS record for xmpp-upload.

Reply:
done!